src/client_state.cpp

```diff
@@ -689,10 +689,12 @@ int wsrep::client_state::leave_toi_local(const wsrep::mutable_buffer& err)
 {
     debug_log_state("leave_toi_local: enter");
     assert(toi_mode_ == m_local);
-    leave_toi_common();
 
+    auto ret = (provider().leave_toi(id_, toi_meta_, err) == provider::success ? 0 : 1);
```

Comment:
Why are these lines swapped? Maybe you could add a comment to clarify?

Reply:
It was " Keep TOI meta data over whole TOI critical section." which didn't make it into PR description. Now updated.

`leave_toi_common()` resets `toi_meta_`, which is now passed to `leave_toi()`.